myia/utils/info.py

```diff
@@ -40,6 +44,8 @@ def top(self):
 def enable_debug():
     """Enable debugging for a context."""
     tok = _debug.set(True)
+    # Clear stack each time we call enable_debug.
+    _stack.clear()
```

Comment:
I think the problem happens because when an exception is raised in `with debug_inherit(...):`, the stack pop does not happen, so a better solution would be to add a `try: finally:` around the `yield` in `debug_inherit`, with the cleanup code in the `finally` block.

Reply:
OK, done. I still have 2 uncovered lines in `myia/ir/print`.